### DISCUSSION TITLE
How to best handle synonyms, acronyms, abbreviations when doing NER with spacy?

### DISCUSSION THREAD

**Discussion by AdrianKrebs** [explosion/spaCy#10753] [Help: Best practices]
What's the best way for handling synonyms, acronyms, and abbreviations of an entity when doing NER? I assume that I can have some kind of lookup table that would then map to the same entity. Let's say for "The North Face" I would want to match all "TNF" as well.

**Comment by polm** [comment]
Depends on your text and what kinds of problems you want to solve - there's no universal approach to this.

Do you have known abbreviations that match specific entities? Then you can use [pattern IDs](https://spacy.io/usage/rule-based-matching#entityruler-ent-ids) to give an ID to each occurrence.

Do you have abbreviations that could map to multiple entities, like ABC for "American Broadcasting Corporation" or "Alcoholic Beverages Commission"? Then you'll want the [Entity Linker](https://github.com/explosion/projects/tree/v3/tutorials/nel_emerson) to disambiguate abbreviations. 

Do you not have abbreviations yet, but need to find them in text? Then maybe you can use [this paper](https://api.semanticscholar.org/CorpusID:28503121)'s method to build a list, or maybe you need a coreference model.
